qemu/tests/usb_host.py

```diff
@@ -67,22 +116,31 @@ def usb_dev_unplug():
                           " returns: '%s'" % reply)
         return
 
-    device = params["usb_host_device"]
+    device = params.get("usb_host_device")
+    if not device:
+        test.cancel("<vendorid>:<productid> of passthrough usb device "
+                    "is not provided.")
     (vendorid, productid) = device.split(":")
+    params["usb_devices"] += " usbhostdev"
+    params["usb_type_usbhostdev"] = "usb-host"
+    params["vendorid_usbhostdev"] = vendorid
+    params["productid_usbhostdev"] = productid
+    params["start_vm"] = 'yes'
+    env_process.preprocess_vm(test, params, env, params["main_vm"])
```

Comment:
Could you explain why do not define these in cfg and boot vm directly ?  Does it used to cancel test case if no vendorid and productid ?

Reply:
IMO, if the `usb_type` is `usb_host`, check if vendorid and productid were defined in the framework is more reasonable.